sklearn/datasets/openml.py

```diff
@@ -424,17 +424,34 @@ def _get_data_qualities(data_id, data_home):
         return None
 
 
-def _get_data_shape(data_qualities):
-    # Using the data_info dictionary from _get_data_info_by_name to extract
-    # the number of samples / features
+def _get_data_instances(data_qualities):
+    """
+    Retrieve the number of instances (samples) from data qualities
+
+    Parameters
+    ----------
+    data_qualities : list
+        used to retrieve the number of instances (samples) in the dataset
+
+    Returns
+    -------
+    instances : int
+        the number of samples in the dataset or -1 if data qualities are
+        unavailable
+    """
+    # If the data qualities are not available, we return -1
+    default_instances = -1
```

Comment:
you directly return the value

```
try:
    return int(float(qualities['NumberOfInstances]))
except KeyError:
    return default_n_samples
else:
    raise

Reply:
As said above, I changed the value retrieval so that's not needed anymore.
Also, the `else` block wouldn't be necessary in this case since the `try` would return before.